packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -248,28 +288,24 @@ export class ForkChoice implements IForkChoice {
       return headBlock;
     }
 
-    const parentBlock = this.protoArray.getBlock(headBlock.parentRoot);
-    const proposalSlot = headBlock.slot + 1;
-    currentSlot = currentSlot ?? this.fcStore.currentSlot;
-
-    // No reorg if parentBlock isn't available
-    if (parentBlock === undefined) {
+    if (headBlock.slot === currentSlot) {
+      // If head block aka the head cache is current, that means `updateHead` is called during gossip handling,
+      // that can only happen if shouldOverrideForkChoiceUpdate = false so no reorg
       return headBlock;
     }
 
-    const {prelimProposerHead} = this.getPreliminaryProposerHead(headBlock, parentBlock, proposalSlot);
+    currentSlot = currentSlot ?? this.fcStore.currentSlot;
 
-    if (prelimProposerHead === headBlock) {
-      return headBlock;
-    }
+    // TODO: Maybe add a transcient flag in ProtoBlock to flag `shouldOverrideFcu` block when importing
+    // so we don't need to call shouldOverrideForkChoiceUpdate again.
```

Comment:
but don't we need to re-evaluate this each time? what's the current scenario in which we don't wanna call `shouldOverrideForkChoiceUpdate` again?

Reply:
I mean, when we import a gossip block, we have already called `shouldOverrideForkChoiceUpdate` to determine if we want to suppress fcu call. But we also call `shouldOverrideForkChoiceUpdate` during `prepareNextSlot` which is redundant. 

`shouldOverrideForkChoiceUpdate` returns the same result on the same block whether it is during the first 4s, or at 8s.

Reply:
> > shouldOverrideForkChoiceUpdate returns the same result on the same block whether it is during the first 4s, or at 8s.
> 
> don't we evaluate single attestations already in fork choice which could change the result?

Oh yea you are right. Forgot about single attestations